src/webpub_manifest_parser/core/ast.py

```diff
@@ -0,0 +1,831 @@
+                self.language,
+                self.sort_as,
+                self.author,
+                self.translator,
+                self.editor,
+                self.artist,
+                self.illustrator,
+                self.letterer,
+                self.penciler,
+                self.colorist,
+                self.inker,
+                self.narrator,
+                self.contributor,
+                self.publisher,
+                self.imprint,
+                self.subject,
+                self.description,
+                self.duration,
+                self.number_of_pages,
+                self.belongs_to,
+            )
+        )
+
+
+class PresentationMetadata(Metadata):
+    """RWPM extension containing presentation metadata."""
+
+    clipped = BooleanProperty("clipped", False)
+    continuous = BooleanProperty("continuous", False)
+    fit = EnumProperty("fit", False, ["width", "height", "contain", "cover"])
+    orientation = EnumProperty("orientation", False, ["auto", "landscape", "portrait"])
+    overflow = EnumProperty(
+        "overflow", False, ["auto", "paginated", "scrolled", "scrolled-continuous"]
+    )
+    spread = EnumProperty("spread", False, ["auto", "both", "none", "landscape"])
+
+
+class CompactCollection(Node):
+    """A compact collection is defined as a grouping of links."""
+
+    links = ArrayOfLinksProperty(key="links", required=True)
+
+    def __init__(self, role=None, links=None):
+        """Initialize a new instance of Collection class.
+
+        :param role: Collection's role (can be empty when self is a manifest)
+        :type role: Optional[CollectionRole]
+
+        :param links: Collection's links
+        :type links: Optional[LinksList]
+        """
+        super(CompactCollection, self).__init__()
+
+        self._role = role
+        self.links = links
+
+    @property
+    def role(self):
+        """Return the collection's role.
+
+        :return: Collection's role.
+        :rtype: Optional[CollectionRole]
+        """
+        return self._role
+
+
+class Collection(CompactCollection):
+    """A collection is defined as a grouping of metadata, links and sub-collections."""
+
+    def __init__(self, role=None, links=None, metadata=None):
+        """Initialize a new instance of Collection class.
+
+        :param role: Collection's role (can be empty when self is a manifest)
+        :type role: Optional[CollectionRole]
+
+        :param links: Collection's links
+        :type links: Optional[LinksList]
+
+        :param metadata: Collection's metadata
+        :type metadata: Optional[Metadata]
+        """
+        super(Collection, self).__init__(role, links)
+
+        self._role = role
+        self._sub_collections = CollectionList()
+        self.metadata = metadata
+
+    metadata = TypeProperty(
+        key="metadata", required=True, nested_type=PresentationMetadata
+    )
+
+    @property
+    def sub_collections(self):
```

Comment:
"subcollection" is idiomatic English so you can drop the `_` I think.

https://www.merriam-webster.com/dictionary/subcollection